### PR TITLE
EvtItem - add support for EntityDropItemEvent

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR aims to add support for `on entity drop` event

Also did some code cleanup as per suggested by Ayham :) 

---
**Target Minecraft Versions:** none
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/events/EvtItem.java [hunk 1]
```diff
@@ -61,10 +62,14 @@ public class EvtItem extends SkriptEvent {
 				.examples("on item spawn of iron sword:",
 						"\tbroadcast \"Someone dropped an iron sword!\"")
 				.since("<i>unknown</i> (before 2.1)");
-		Skript.registerEvent("Drop", EvtItem.class, PlayerDropItemEvent.class, "[player] drop[ing] [[of] %-itemtypes%]")
-				.description("Called when a player drops an item from their inventory.")
-				.examples("on drop:")
-				.since("<i>unknown</i> (before 2.1)");
+		Skript.registerEvent("Drop", EvtItem.class, CollectionUtils.array(PlayerDropItemEvent.class, EntityDropItemEvent.class),
+				"[(player|1¦entity)] drop[ing] [[of] %-itemtypes%]")
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/events/EvtItem.java:66]:
```suggestion
				"[player|1:entity] drop[ing] [[of] %-itemtypes%]")
```

File: src/main/java/ch/njol/skript/events/EvtItem.java [hunk 2]
```diff
@@ -61,10 +62,14 @@ public class EvtItem extends SkriptEvent {
...
+				.description("Called when a player drops an item from their inventory, or an entity drops an item, such as a chicken laying an egg.")
+				.examples("on drop:",
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/events/EvtItem.java:68]:
```suggestion
				.examples("on drop:",
						"\tif event-item is compass:",
						"\t\tcancel event",
						"",
```

File: src/main/java/ch/njol/skript/events/EvtItem.java [hunk 3]
```diff
@@ -61,10 +62,14 @@ public class EvtItem extends SkriptEvent {
...
+						"on entity drop an egg:",
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/events/EvtItem.java:69]: Sounds better \:)
```suggestion
						"on entity drop of an egg:",
```

File: src/main/java/ch/njol/skript/events/EvtItem.java [hunk 4]
```diff
@@ -140,6 +145,8 @@ public boolean check(final Event e) {
 			EffSecSpawn.lastSpawned = ((ItemSpawnEvent) e).getEntity();
 		if (hasEntityPickupItemEvent && ((!entity && e instanceof EntityPickupItemEvent) || (entity && e instanceof PlayerPickupItemEvent)))
 			return false;
+		if ((!entity && e instanceof EntityDropItemEvent) || (entity && e instanceof PlayerDropItemEvent))
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/events/EvtItem.java:148]: Change `e` to `event`

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
⚡

File: src/main/java/ch/njol/skript/events/EvtItem.java (hunk 4 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/events/EvtItem.java]: done

File: src/main/java/ch/njol/skript/events/EvtItem.java [hunk 5]
```diff
@@ -61,10 +62,17 @@ public class EvtItem extends SkriptEvent {
 				.examples("on item spawn of iron sword:",
 						"\tbroadcast \"Someone dropped an iron sword!\"")
 				.since("<i>unknown</i> (before 2.1)");
-		Skript.registerEvent("Drop", EvtItem.class, PlayerDropItemEvent.class, "[player] drop[ing] [[of] %-itemtypes%]")
-				.description("Called when a player drops an item from their inventory.")
-				.examples("on drop:")
-				.since("<i>unknown</i> (before 2.1)");
+		Skript.registerEvent("Drop", EvtItem.class, CollectionUtils.array(PlayerDropItemEvent.class, EntityDropItemEvent.class),
+				"[player|1:entity] drop[ing] [[of] %-itemtypes%]")
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/events/EvtItem.java:66]:
```suggestion
				"[player|1:entity] drop[ping] [[of] %-itemtypes%]")
```
The gerund of "drop" is "dropping" but that pattern allows for "droping" which is not correct

**Review by UnderscoreTud — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/events/EvtItem.java (hunk 5 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/events/EvtItem.java:66]: jebus, how did I screw that one up? haha, oops! thanks for that!

**Review by AyhamAl-Ali — APPROVED** [review]

**Pull request merged**